pyveda/vedaset/store/vedabase.py

```diff
@@ -103,9 +103,9 @@ class H5DataBase(BaseDataSet):
     """
     _sample_class = H5SampleArray
     _variable_class = H5VariableArray
-    _frozen = ("mltype", "image_shape", "image_dtype", "classes")
+    _frozen = ("mltype", "image_shape", "image_dtype", "classes", "dataset_id")
 
-    def __init__(self, fname, title="SBWM", overwrite=False, mode="a", **kwargs):
+    def __init__(self, fname, dataset_id, title="SBWM", overwrite=False, mode="a", **kwargs):
```

Comment:
This makes it so that we have to pass in some value for dataset_id, but we don't always necessarily have a corresponding dataset_id if we're just working with data locally or that isn't already referenced in Veda. We could set default like `dataset_id=None`, but I'm not sure that pytables supports `NoneType` attributes (you should check this). We could otherwise just do something like `dataset_id="Undefined"` or something for the time being